compiler/compiler.go

```diff
@@ -185,11 +186,17 @@ func NewTargetMachine(config *Config) (llvm.TargetMachine, error) {
 	if err != nil {
 		return llvm.TargetMachine{}, err
 	}
-	features := strings.Join(config.Features, ",")
 
+	var features string
 	var codeModel llvm.CodeModel
 	var relocationModel llvm.RelocMode
 
+	if strings.HasPrefix(config.Triple, "wasm") {
+		features = config.WasmFeatures
+	} else {
+		features = strings.Join(config.Features, ",")
```

Comment:
This asymmetry bothers me a bit. I was thinking about whether we can move the join call when we retrieve the features but decided to keep it in case people didn't like it.

Reply:
Did you really mean to disable the "normal" features if `wasm-features` is being used? Seems like there is probably a case where you need both. Perhaps this is enough here (untested):

```go
features := strings.Join(config.Features, ",")
if strings.HasPrefix(config.Triple, "wasm") && len(config.WasmFeatures) > 0 {
        features = strings.Join([]string{features, config.WasmFeatures}, ",")
}
```

Reply:
Actually the above has flaw. This code here is what I *really* meant :smile_cat: 

```go
feat := config.Features
if strings.HasPrefix(config.Triple, "wasm") && len(config.WasmFeatures) > 0 {
        feat = append(feat, config.WasmFeatures)
}
features := strings.Join(feat, ",")
```

Reply:
@deadprogram Yes, that was on purpose.  I can change the code so both are taking into account but I don't think is needed at the moment and would be easy to change it later.

Reply:
As discussed offline, I've added your suggestion. Thank you.